routing/probability_bimodal.go

```diff
@@ -484,43 +488,57 @@ func (p *BimodalEstimator) probabilityFormula(capacityMsat, successAmountMsat,
 		return 0.0, nil
 	}
 
-	// Mission control may have some outdated values, we correct them here.
-	// TODO(bitromortac): there may be better decisions to make in these
-	//  cases, e.g., resetting failAmount=cap and successAmount=0.
-
-	// failAmount should be capacity at max.
-	if failAmount > capacity {
-		log.Debugf("Correcting failAmount %v to capacity %v",
-			failAmount, capacity)
+	// The next statement is a safety check against an illogical condition.
+	// We discard the knowledge for the channel in that case since we have
+	// inconsistent data.
+	if failAmount <= successAmount {
+		log.Warnf("Fail amount (%s) is smaller than or equal to the "+
+			"success amount (%s) for capacity (%s)",
+			failAmountMsat, successAmountMsat, capacityMsat)
 
+		successAmount = 0
 		failAmount = capacity
 	}
 
-	// successAmount should be capacity at max.
-	if successAmount > capacity {
-		log.Debugf("Correcting successAmount %v to capacity %v",
-			successAmount, capacity)
-
-		successAmount = capacity
+	// Mission control may have some outdated values with regard to the
+	// current channel capacity between a node pair. This can happen in case
+	// a large parallel channel was closed or if a channel was downscaled
+	// and can lead to success and/or failure amounts to be out of the range
+	// [0, capacity]. We assume that the liquidity situation of the channel
+	// is similar as before due to flow bias.
+
+	// In case we have a large success we need to correct it to be in the
+	// valid range. We set the success amount close to the capacity, because
+	// we assume to still be able to send. Any possible failure (that must
+	// in this case be larger than the capacity) is corrected as well.
+	if successAmount >= capacity {
+		log.Debugf("Correcting success amount %s and failure amount "+
+			"%s to capacity %s", successAmountMsat,
+			failAmount, capacityMsat)
+
+		successAmount = capacity - 1
```

Comment:
hmm what's the reason for minus 1?

Reply:
so the successAmt needs to be -1 smaller for the reNorm calculation to not be 0 basically ?

Reply:
> so the successAmt needs to be -1 smaller for the reNorm calculation to not be 0 basically ?

right, I think that's another way to say that there would be a logical contradiction to say that you can send the same amount as you fail to send